[feat] engine: add right dao

0xhtml opened the pull request:
## What does this PR do?

Add Right Dao as an engine.
https://rightdao.com

## Why is this change important?

Adding new engine.
Information on engine: https://seirdy.one/posts/2021/03/10/search-engines-with-own-indexes/ (ctrl-f for right dao)

## How to test this PR locally?

query `!rd …`

0xhtml committed:
[feat] engine: add right dao

Bnyro commented on searx/settings.yml:
```diff
@@ -1548,6 +1548,24 @@ engines:
+    search_url: https://rightdao.com/search?q={query}&start={pageno}
+    results_xpath: //div[@class="description"]
+    url_xpath: ../div[@class="title"]/a/@href
+    title_xpath: ../div[@class="title"]
```
```suggestion
    results_xpath: //div[contains(@class, "description")]
    url_xpath: ../div[contains(@class, "title")]/a/@href
    title_xpath: ../div[contains(@class, "title")]
```
prevents in from breaking in case they add additional classes to these html elements.

Bnyro commented on searx/settings.yml:
```diff
@@ -1548,6 +1548,24 @@ engines:
+    paging: true
+    page_size: 12
+    search_url: https://rightdao.com/search?q={query}&start={pageno}
+    results_xpath: //div[@class="description"]
```
Why don't you use `//div[contains(@class, "item")]` here to directly match the parent element? That way you wouldn't need to use `../` below, would you?

0xhtml replied:
The class item is also used for headers and news stories in the results. Only genuine results have a description.

0xhtml committed:
[enh] right dao engine: allow additional classes

Allow additional classes while parsing to prevent the engine from breaking in the future if additional classes are added to the elements.

Co-authored-by: Bnyro <bnyro@tutanota.com>
